lib/masamune/schema/map.rb

```diff
@@ -43,12 +43,36 @@ def skip?
         @store.json_encoding == :quoted
       end
 
-      def encode(fields = [])
-        fields.map { |field| field =~ /^{|}$/ ? quote(field) : field }
+      def encode(line, separator)
+        fields = []
+        buffer = ''
+        nested = false
+        line.chomp.each_char do |char|
```

Comment:
Use strip unless you really mean to get rid of the last character rather than just whitespace.